### ISSUE TITLE
Assets

### ISSUE THREAD

**Issue by plazatin** [DirtyHairy/cloudpilot#2]
Hi Christian, here's a clearer silkscreen image for the palm v, in svg format.

![icons_silkscreen-palmv](https://user-images.githubusercontent.com/64849293/108872912-a21bb080-7635-11eb-8616-56ccb64753be.png)

[icons_palm-v.zip](https://github.com/DirtyHairy/cloudpilot/files/6030304/icons_palm-v.zip)

**Comment by DirtyHairy** [comment]
Hi Paolo!

Thanks alot, looks awesome. Already integrated into the latest builds :smirk:

**Comment by plazatin** [comment]
Here's an altered one based on the m515, in case you ever decide to use different silkscreens depending on the rom or user preference.

![silkscreen-m515-bwpng](https://user-images.githubusercontent.com/64849293/109124619-16fe0000-7786-11eb-8ecd-76951026e01b.png)

![silkscreen-m515](https://user-images.githubusercontent.com/64849293/109123107-6ba07b80-7784-11eb-8e0b-5811d5982890.png)

[silkscreen-m515-bw.zip](https://github.com/DirtyHairy/cloudpilot/files/6041535/silkscreen-m515-bw.zip)
[silkscreen-m515.zip](https://github.com/DirtyHairy/cloudpilot/files/6041444/silkscreen-m515.zip)

**Comment by DirtyHairy** [comment]
Thanks alot, added...:

<img width="430" alt="Bildschirmfoto 2021-02-25 um 22 59 11" src="https://user-images.githubusercontent.com/2444224/109225274-55fb7800-77bd-11eb-8400-268c09363d91.png">

**Comment by plazatin** [comment]
I also made hardware buttons based on the two currently supported devices, the m515 and palm v, again in svg format (with PNGs in case you prefer that).

[hbuttons-palm.zip](https://github.com/DirtyHairy/cloudpilot/files/6053894/hbuttons-palm.zip)

![hbutton-preview-m515](https://user-images.githubusercontent.com/64849293/109380902-3de93d00-7912-11eb-85da-f9583710ef20.png)

![hbutton-preview-palmv](https://user-images.githubusercontent.com/64849293/109380919-4d688600-7912-11eb-995d-5cdf973f619e.png)

Each button is a single image at the moment, and the rockers too. This can easily be split or combined to one whole image similar to the silkscreen. I haven't really taken a look at how you implemented the buttons/how you intend to implement the buttons, or if this is even the route you were looking to take. Maybe you'd prefer to have simpler and/or original icons uniform across the devices?

**Comment by DirtyHairy** [comment]
Actually, those look amazing, and I'll definitely be using them, thanks alot! The size of the individual tiles is perfect, too, as is SVG. I have only one change: could you maybe combine the buttons for each device into a single 480x90 SVG (just like your mockups)?

**Comment by plazatin** [comment]
Sure, here you go, single image for each. I wasn't sure if you wanted a pushed down button on click so I separated them initially.

I also adjusted the size of the m515 rocker similar to what I did with the palm V.

[hbuttons-palm-r2.zip](https://github.com/DirtyHairy/cloudpilot/files/6059480/hbuttons-palm-r2.zip)

![hard-buttons-palmv](https://user-images.githubusercontent.com/64849293/109458925-ba0d8d00-7a98-11eb-8f09-a3250870003e.png)
![hbuttons-m515](https://user-images.githubusercontent.com/64849293/109458943-bf6ad780-7a98-11eb-874d-e9417368246b.png)

**Comment by DirtyHairy** [comment]
Added. I am drawing tranlucent recangles over the the active areas of the buttons that are currently touched in order to visualize the interaction and the active region.

Palm m515:

![IMG_1006](https://user-images.githubusercontent.com/2444224/109489232-d86a8d00-7a86-11eb-90e4-4f3749a86ab0.jpeg)

Palm V:

![IMG_1007](https://user-images.githubusercontent.com/2444224/109489240-dacce700-7a86-11eb-957a-a4ffd0ff7124.jpeg)

What do you think? If you like you can also try the app on (https://dirtyhairy.github.io/cloudpilot/app), it is almost fully functional now. The only bigger pieces that are still missing are image exports, hotsync name management (together with a proper settings dialog) and the service worker.

**Comment by DirtyHairy** [comment]
Ouch, sorry for the massive screenshots 😏

**Comment by plazatin** [comment]
Looking great! I was initially bothered about the overlay on click (since it highlighted the entire block not just the button), but it actually grew on me as I played with it longer. I need the visual feedback. With a "clicked button" animation, the click feedback might not be visible if the finger is over a button, especially with the rocker. A clicking sound could help, but personally I always keep my phone on silent. So it seems fine as it is.

Another thing that I was initially looking out for was misclicks between the down button of the rocker and the sessions tab, but seems like the occurrence was pretty rare, especially on my phone (bigger resolution so there was more space between them). Had to actually put an effort for it to happen. Maybe other people could share their experiences with it.

One comment I do have is the plus button. Not sure if the icons are temporary but I think it could benefit to change the install package since it's the same icon and same spot as new session when on the sessions tab. Maybe use something like the update icon for installing packages to reduce confusion?

![update-icon-ios](https://user-images.githubusercontent.com/64849293/109502807-6e78d480-7ad4-11eb-9554-1d1f37693593.PNG)

**Comment by plazatin** [comment]
By the way, the hardware buttons svg doesn't show up on firefox. I think you had and fixed this issue for the silkscreen

**Comment by DirtyHairy** [comment]
> A clicking sound could help, but personally I always keep my phone on silent. So it seems fine as it is.

Same with me 😏 Also, I plan to add audio emulation after the PWA is finished, and I don't think that would go well with audible button clicks.

> Another thing that I was initially looking out for was misclicks between the down button of the rocker and the sessions tab

Och. You are right, I didn't think about that. I also manage to hit "search" instead of notes from time to time. I'll ask for feedback specifically on this when the first iteration of the PWA is done. I don't have a good idea how to solve this, though.

> Maybe use something like the update icon for installing packages to reduce confusion?

Good point, thanks. Actually, have the same icon for both has bothered me too. I have implemented your suggestion.

> By the way, the hardware buttons svg doesn't show up on firefox. I think you had and fixed this issue for the silkscreen

Indeed. Thanks, fixed 😏

**Comment by plazatin** [comment]
I updated the silkscreen to reduce the misclicks between search and notes (they were too close to the edges and the hardware buttons so I moved them towards the center a bit). That should already reduce misclicks from a visual standpoint, but you can also adjust the rectangle hotspots if needed.

![silkscreen-m515](https://user-images.githubusercontent.com/64849293/109586260-dadced80-7b3f-11eb-9e0c-c478e3fb94c5.png)
![silkscreen-palmv](https://user-images.githubusercontent.com/64849293/109586262-db758400-7b3f-11eb-99ae-7dc1ebfc6ec5.png)
![silkscreen-palmv-bw](https://user-images.githubusercontent.com/64849293/109586265-dc0e1a80-7b3f-11eb-939a-8547267377a0.png)

[silkscreen-palm-v2.zip](https://github.com/DirtyHairy/cloudpilot/files/6065701/silkscreen-palm-v2.zip)

Preview:
![preview-adjusted-silkscreen](https://user-images.githubusercontent.com/64849293/109587598-5049bd80-7b42-11eb-8cd1-355162713dc5.png)

I added the green version of the silkscreen palmv just in case. I also applied the px dimensions to the svg already, if I'm not mistaken that was the fix for the firefox rendering issue.

We can still further reduce the misclicks by making some adjustments to the layout, I have some ideas and will come up with a mockup when I find the time.

**Comment by plazatin** [comment]
> 
> Good point, thanks. Actually, have the same icon for both has bothered me too. I have implemented your suggestion.
> 

Looks a lot less confusing now, thanks!

**Comment by DirtyHairy** [comment]
Thanks alot.

> That should already reduce misclicks from a visual standpoint, but you can also adjust the rectangle hotspots if needed.

I am afraid this is not possible --- the silkscreen is managed by PalmOS. From the perspective of the emulator, the whole canvas (apart from the button area) is just a 160x220 touchscreen.

However, what it be possible to disable the tab icons and the lower part of the silkscreen for a short time (500msec or so) after the hardware buttons have been interacted with (or maybe even map that part of the silkscreen to the button area). Maybe not on a simple touch like opening the notepad, but after there have been interactions for an extended period of time.

**Comment by DirtyHairy** [comment]
I added the new silkscreens. I am sticking with the B/W one for the Palm V for now --- the green one would require me to adjust the palette to look good.

**Comment by plazatin** [comment]
Hi Christian, sorry it took a while, been busy with other stuff. 

After playing around with the emulator more, I realized I don't experience accidental clicks anymore, even with the two closest and tightest buttons, the down rocker and sessions. I'm not sure if I attribute that to the additional space in between them (black bar due to my device's resolution) or just getting used to the layout. That said, I had a few ideas a few weeks back to prevent this and I'll share them just the same. It might still be worth a look in case others have different experiences or in case you want to experiment with them.

The few accidental clicks I had were all from the bottom, so I figured it was the bottom that needed changes. The first idea I had was to get rid of the bottom altogether and add the upper portion of the "device". 

![mockup-ui-A-1](https://user-images.githubusercontent.com/64849293/110236247-b59d1480-7f6f-11eb-8e83-52d0b4b52a43.png)

Mockup A:
- Rough mockup, would have the logo, the model emulated, and a led light (for this example I remembered the treo) that would function as the savestate status (green okay, red error, blank temporary issues).  The rest of the ui is copied from mobile browsers. The power is on the upper left side where it would be safe from accidental presses. In all the mockups, I thought of combining the power button with the reset buttons, similar to how windows does it. To its right is the session name, and next to it is the sessions button (same as the browser tabs bar), which indicates the number of sessions. To its right is the kebab menu. Since it is copied from browsers, people would be familiar with it.

Mockup B:
- Same as A, with the slight difference of not showing the sesions button. The top center has the sessions name with a bit more info on the session such as the deviec, ram and OS.  Clicking on this will go to the sessions page. With no session running, this can just be a plus button or text that says something like click to start or new session.

![mockup-ui-C-D](https://user-images.githubusercontent.com/64849293/110236836-05310f80-7f73-11eb-851c-74f8f3304df2.png)

Mockup C:
- Has the App name and/or logo at the center. The bottom part isn't removed but replaced with sessions. To its right, the check for savestate status that you initially had. Since the only button at the bottom is the tabs/sessions button, app buttons being too close to the hardware buttons will no longer be an issue.

Mockup D:
- Same as C, except there is no sessions button, clicking on the sessions text will open the sessions page (or you can actually go extreme and have the sessions in the kebab menu. This means the bottom part will entirely be for display/read-only, no more accidental clicks). The top center is just the logo, so we can put back the install app if you prefer as it's not too crowded.

![mockup-ui-sessions](https://user-images.githubusercontent.com/64849293/110237043-a371a500-7f74-11eb-825d-4a8143096c2b.png)
In the sessions page and settings page, the back is at the top.

The rest of the stuff can accessed on the kebab menu at the upper right.

Personally, I like the top portion of D with the bottom portion of C. I would also be fine if the current tab buttons stayed, as I no longer experience accidental clicks.

**Comment by DirtyHairy** [comment]
Hi Paolo!

Thanks for your mockups and ideas. Although I'd like to stick with the tab layout for the time being, there's some cool ideas in there that we might reconsider at some later point. In particular, I like the skeuomorph bar above the LCD in your mockups A and B. I have silently axed the indicator from the emulator, though, I figured it was too technical and shouldn't be necessary anyway. When there is a fatal flaw saving the emulator stops, and a modal dialog offers the option to reload and optionally to save an image file first.

So far, I have been doing more or less an iOS design, but your mockups have made me remember that Android has the navbar at the bottom that gobbles up space, and I am slightly concerned that the tabbar might consume too much space on devices with a small screen. I have made a change that automatically shrinks the tabbar by 50% on the emulation page if this benefits the canvas size.

Functionally, the app is done now, what remains is fine tuning and some technical debt.
